tests/test_client/tests.py

```diff
@@ -85,6 +87,47 @@ def test_post(self):
         self.assertEqual(response.templates[0].name, 'POST Template')
         self.assertContains(response, 'Data received')
 
+    def assertJSONResponse(self, method, data, method_name):
+        """Assert an appropriate response was received from a JSON request."""
+
+        response = method('/json_view/', data, content_type=JSON_CONTENT)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.context['data'], 37)
+        self.assertEqual(response.templates[0].name, '{} Template'.format(method_name))
+        self.assertContains(response, 'Viewing {} page.'.format(method_name))
+
+    def test_json_post(self):
+        """POST some JSON data to a view"""
+        self.assertJSONResponse(self.client.post, {'value': 37}, 'POST')
+
+    def test_json_put(self):
+        """PUT some JSON data to a view"""
+        self.assertJSONResponse(self.client.put, {'value': 37}, 'PUT')
+
+    def test_json_patch(self):
+        """PATCH some JSON data to a view"""
+        self.assertJSONResponse(self.client.patch, {'value': 37}, 'PATCH')
+
+    def test_json_delete(self):
+        """DELETE some JSON data to a view"""
+        self.assertJSONResponse(self.client.delete, {'value': 37}, 'DELETE')
+
+    def test_encoded_json_post(self):
```

Comment:
I wouldn't include this test in the PR because it's testing existing behavior. But it doesn't seem needed as there's a test in `test_client_regress` that fails if the `isinstance(data, dict)` check is `_encode_json` is removed.